detox/android/detox/src/full/java/com/wix/detox/reactnative/idlingresources/timers/FabricTimersIdlingResource.kt

```diff
@@ -0,0 +1,67 @@
+package com.wix.detox.reactnative.idlingresources.timers
+
+import android.view.Choreographer
+import androidx.test.espresso.IdlingResource
+import com.facebook.react.bridge.ReactContext
+import com.facebook.react.modules.core.JavaTimerManager
+import com.wix.detox.reactnative.idlingresources.DetoxIdlingResource
+import org.joor.Reflect
+import kotlin.reflect.full.declaredFunctions
+import kotlin.reflect.jvm.isAccessible
+
+class FabricTimersIdlingResource(
+    private val reactContext: ReactContext,
+    private val getChoreographer: () -> Choreographer = { Choreographer.getInstance() }
+) : DetoxIdlingResource(), Choreographer.FrameCallback {
+
+    override fun checkIdle(): Boolean {
+        val hasActiveTimers: Boolean = hasActiveTimers()
+
+        if (hasActiveTimers) {
+            getChoreographer().postFrameCallback(this@FabricTimersIdlingResource)
```

Comment:
Future note (todo?): Add ourselves as a callback in the timers module, in-sync with the longest current wait in queue*. Brut-force checking every 16ms was not the best idea, in retrospect.

_\* still below the window threshold..._